test/performance/odp_timer_accuracy.c

```diff
@@ -1263,34 +1476,60 @@ static int run_test(void *arg)
 			}
 
 			if (ret != ODP_TIMER_SUCCESS) {
-				printf("Timer set failed: %i. Timeout nsec "
-				       "%" PRIu64 "\n", ret, ctx->nsec);
+				ODPH_ERR("Timer set failed: %i. Timeout nsec "
+					 "%" PRIu64 "\n",
+					 ret, ctx->nsec);
 				return 0;
 			}
 		} else if (mode == MODE_PERIODIC) {
 			int ret = odp_timer_periodic_ack(ctx->timer, ev);
 
 			if (ret < 0)
-				printf("Failed to ack a periodic timer.\n");
+				ODPH_ERR("Failed to ack a periodic timer.\n");
 
 			if (ret == 2)
 				odp_atomic_inc_u64(&test_global->last_events);
 
 			if (ret == 2 || ret < 0)
 				odp_event_free(ev);
+		} else if (mode == MODE_CONCURRENCY && events < tot_timers - 1) {
+			process_event_concurrency(events, test_global, ctx, time_ns, ev);
 		} else {
 			odp_event_free(ev);
 		}
 	}
 
 	if (test_global->opt.groups) {
 		if (odp_schedule_group_leave(group, &mask))
-			printf("odp_schedule_group_leave() failed\n");
+			ODPH_ERR("odp_schedule_group_leave() failed\n");
 	}
 
 	return 0;
 }
 
+static void interval_loop_concurrency(test_global_t *test_global)
+{
+	uint64_t events_prev = 0;
+	uint64_t events = 0;
+	uint64_t start_ns = odp_time_global_strict_ns();
+	uint64_t prev_ns = start_ns;
+
+	while ((events = odp_atomic_load_u64(&test_global->events)) < test_global->opt.tot_timers) {
+		odp_time_wait_ns(test_global->opt.interval * ODP_TIME_SEC_IN_NS);
```

Comment:
Because `test_global->events` value is read before wait, the first print round results aren't correct. At least with larger interval values this can be confusing. E.g.
```
Starting timers took 776 nsec
sec 60 total events 0 events 0 events/s 0
sec 120 total events 54545 events 54545 events/s 909

```

Reply:
I forgot about that. Fixed in v4.